rlclientlib/logger/async_batcher.h

```diff
@@ -151,11 +156,28 @@ namespace reinforcement_learning { namespace logger {
       }
     }
 
-    RETURN_IF_FAIL(collection_serializer.finalize(status));
-
+    if (_events_counter_status == events_counter_status::ENABLE) {
+      uint64_t buffer_start_event_number = _buffer_end_event_number;
```

Comment:
Can we change buffer_start_event_number to buffer_start_event_index and _buffer_end_event_number to _buffer_end_event_index? You are accessing _event_index.